enhancements/installer/aws-custom-dns.md

```diff
@@ -0,0 +1,476 @@
+`api-int.<cluster domain>`, `*.apps.<cluster domain>` and `*.<cluster domain>`.
+The cloud Load Balancer is still the preferred solution for providing load
+balancing services for `api`, `api-int` and `ingress` in public and private
+hosted zones.
+
+### User Stories
+
+- As an administrator, I would like to deploy OpenShift 4 clusters to supported
+cloud providers leveraging my custom DNS service.
+
+### Goals
+
+- Enable AWS customers to use their custom DNS solution instead of the cloud
+solution (Route53).
+- Lay the groundwork to add this capability for other cloud platforms like
+Azure and GCP.
+- Continue using the cloud based Load Balancer service for API, Ingress and
+API-Int IPs.
+
+### Non-Goals
+
+- We will not focus on Installer changes to support the custom DNS solution on
+Azure and GCP platforms. That will come in follow-up work and will rely on the
+changes introduced within this solution.
+
+- We will not use the [Kubernetes External DNS provider](https://github.com/openshift/external-dns) and/or the [OpenShift External DNS Operator](https://github.com/openshift/external-dns-operator) to program dns entries across varied DNS providers.
+The External DNS Operator relies on being able to program the customer's
+DNS solution. The customer DNS solutions in consideration here do not allow
+OpenShift to configure them.
+
+## Proposal
+
+For the OpenShift installation to make use of a custom DNS solution that is not
+configurable by OpenShift, the customer is responsible for configuring the
+DNS entries for the API and Ingress services.
+
+When it is indicated to the OpenShift installer via an input parameter that the
+custom DNS solution is in use, OpenShift does not configure the cloud DNS
+solution [Route53 for AWS](https://aws.amazon.com/route53/). OpenShift will
+continue to configure and manage the cloud load balancing solution [Elastic
+Load Balancer](https://aws.amazon.com/elasticloadbalancing/) for all its
+services.
+
+OpenShift will start static CoreDNS pods to provide DNS resolution for API,
+Ingress API and Ingress services that are essentail for cluster creation.
+Although not identical, this approach leverages learning from OpenShift's
+approach to providing these services for on-prem platforms.
+
+### Workflow Description
+
+1. For an OpenShift administrator to use a custom DNS solution, they would
+have to enable a config option in the install configuration. This would
+cause the Installer and the Ingress Operator to skip configuring the cloud
+DNS with entries for API, Internal API and Ingress Services. 
+
+2. The Installer and the Ingress Operator will continue to configure the cloud
+Load Balancers on the public subnets of Public Hosted Zones for `api` and
+`*.apps`. They will continue to configure cloud Load Balancers on the private
+subnets of Private Hosted Zones for `api-int` and `*.apps` for in-cluster
+access to the Ingress service.
+
+3. The Installer also creates a ConfigMap(CloudLBConfig) to store LB API
```

Comment:
This (creating an in-cluster resource after terraform execution) is sort of uncharted territory for the installer, so I think it would be beneficial to think through exactly how/when this is created.

Reply:
The CloudLBConfing ConfigMap is serving as the location for storing these IP addresses in this proposal. Yes, we could use the PlatformStatus to pass this information as in the on-prem case. In the custom DNS case, we want to provide these addresses in a friendly format to the customer to configure their own DNS. If PlatformStatus can do the job then we could just use that instead of the ConfigMap.